### DISCUSSION TITLE
Improvements

### DISCUSSION THREAD

**Discussion by Electrik-rich546456** [NightscoutFoundation/xDrip#3740] [Ideas]
Hello Ive been using xdrip to receive the data from my FSL2 sensor for a long time now and always thought it was silly to have the need of OOP2 to decode the data from the sensor. Considering this [app](https://github.com/j-kaltes/Juggluco) can do it, why can't xdrip?
Am i mistaken by the looks of your comment https://github.com/NightscoutFoundation/xDrip/pull/1524#issuecomment-737964226 .

Does anyone know if there is a branch that has the ability so send the data gathered to libreview for my health provider to see my levels and cards and insulin administered

**Comment by Teute0815** [comment]
- You can use juggluco to receive your sensor data (**glucose values**) and forward it to xDrip via different means (libre broadcast, local Nightscout, Eversense)
- In juggluco you can also add your **LibreView** account, so all data will be uploaded there, if you want.
- Regarding **insulin**, you would obviously have to put that it into juggluco in order to receive it in LibreView, because xDrip doesn't connect to any sensor vendor for good reasons [#2742](https://github.com/NightscoutFoundation/xDrip/discussions/2742#discussioncomment-5825906)

If it doesn't have to be an original/official "LibreView" Documentation, you can retrieve glucose values from juggluco, add treatments in xDrip and send all this data to nightscout and have a report tool (like Nightscout Reporter) generate your reports - that's how I do it.

Libre 3 -> juggluco (glucose values) -> Libre patched/Local webserver/Eversense -> xDrip (retrieved glucose values + added treatments) -> Nightscout -> [Nightscout Reporter](https://nightrep.zreptil.de/) -> generate various in depth reports

Some people are also asking for a fully filled OEM LibreLink app. At least for the glucose values you can achieve that by force stopping juggluco half an hour before sensor expiry and scan the sensor with the OEM app. That way you get all the sensor data into your app (no insulin, though). Force stop OEM and rescan with juggluco to get back to your original setting.
In order to be able to do that, you have to retrieve your account ID from LibreView into juggluco (Settings/LibreView) otherwise the sensor won't be recognized. Retrivieng your AccountID is a "one time" thing, you won't have to be connected/sending to LibreView all the time in order to get that working.

**Reply by Electrik-rich546456** [reply]
Hmm i still don't understand why juggluco can send data into Libreview  and xdrip cant ? 
Breach the terms and conditions of what exactly? I didn't sign up to have this debilitating disease and with the whole WeAreNotWaiting movement I honestly think any company that help should let us be and do everything in there powers to make life easier for us all and periodic NFC scans by Librelink is definitely not the correct way on the other hand minute by minute data points and insulin amounts would help a lot of people out. I for one would appreciate my health professional being able to see it all but they use Librelink just making the situation worse.

**Reply by parapenT1sta** [reply]
To avoid legal issues with Abbott...

**Reply by Electrik-rich546456** [reply]
> To avoid legal issues with Abbott...

If that is the case how and why is juggluco able to get away with it ???

**Reply by parapenT1sta** [reply]
In the past it happened with the abbott patched app, it might happe again, we never know: https://boingboing.net/2019/12/12/they-literally-own-you.html
https://www.diabettech.com/wearenotwaiting/patching-librelink-for-libre2-clearing-the-fud/

Here is the opinion from @jamorham: https://github.com/NightscoutFoundation/xDrip/discussions/2118#discussioncomment-3909939

**Reply by PhilTU** [reply]
> > To avoid legal issues with Abbott...
> 
> If that is the case how and why is juggluco able to get away with it ???

you get away with it until you can't. Juggluco was pulled from the play store. He uses a lot of Librelink in his app (used to be a prerequisite that you had the offficial app or an apk for it to suck it out of).

xDrip+ is insulated from any Abbott histrionics by having a separate helper app.

**Reply by Electrik-rich546456** [reply]
I was not aware and very shocked of the take down notices in the past that really sucks.
Its doesnt really help to bring diabetics one step closer to a better quality of life.

Are you talking about oop2 @PhilTU ?
If you are then same question again, How does that work? i believe  oop2 is not open source so if that is the reason why oop2 has not been handed a takedown notice  then why can't we reverse engineer that situation and somehow close the source and add code to decode fsl 1 and 2 and maybe 3 and send it to librelink site for the health professionals and then Abbott would not be able to complain about it?

**Reply by PhilTU** [reply]
I was referring to Juggluco's use of Abbott code. That is a vulnerability in terms of DMCS action by Abbott. By keeping sensor specific vendor code or algorithms out of xDrip the project is insulated from this.

Accessing Libreview contrary to the terms & conditions adds to the exposure, Diabox had to remove its Libreview upload.

So while you might find a helper app like OOP2 "silly" it provides a degree of protection and can also be chsnged quickly in response to new sensor types.

Juggluco web site could be shut down tomorrow morning.

**Reply by Electrik-rich546456** [reply]
Ok so how does oop2 get away with it then ?

**Reply by PhilTU** [reply]
Either it doesn't breach any IP or it hasn't been noticed. Glimp would be another example.

**Reply by Electrik-rich546456** [reply]
Glimp doesn't work with fsl2...
as regards oop2 not getting DMCS how and why can't the good apps just have it built in ?

**Reply by PhilTU** [reply]
Did you try Glimp on a Libre 2 ;-)

some apps contain things that can be interpreted as breaking the law by infringing someone's rights. OOP2 apparently does not. So it's more a case of what's not in OOP2 than what is. 

Can you install J..lo without having Libre...k present ? Why is Dia..x apk 50 MB ?
